libs/nx-remotecache-s3/src/lib/setup-s3-task-runner.ts

```diff
@@ -75,19 +73,25 @@ export const setupS3TaskRunner = async (
         /* eslint-disable @typescript-eslint/naming-convention */
         Bucket: bucket,
         Key: `${prefix}${filename}`,
+
         /* eslint-enable @typescript-eslint/naming-convention */
       });
-      return getStream.buffer(result.Body as Stream);
+
+      if (typeof result.Body === 'undefined') {
+        throw new Error('Could not retrieve file');
+      }
+
+      return result.Body.transformToWebStream() as unknown as NodeJS.ReadableStream;
     },
-    storeFile: async (filename: string, buffer) => {
+    storeFile: async (filename: string, stream) => {
       if (readOnly) {
         throw new Error('ReadOnly');
       }
       return await s3Storage.putObject({
         /* eslint-disable @typescript-eslint/naming-convention */
         Bucket: bucket,
         Key: `${prefix}${filename}`,
-        Body: buffer,
+        Body: stream,
```

Comment:
Actually this is not working, I get the following error:
```
Warning: Failed to store cache to S3
File: 2e6242eb3a522125b26a34409b3acbae9e7f406eb515f2fd25765cfa51c5ad79.tar.gz
Error: Missing header: Content-Length
```

It seems that the Content-Length header becomes mandatory once we upload a Readable through putObject: https://github.com/aws/aws-sdk-js-v3/issues/2348

Reply:
Oh interesting indeed! It seems we may need to keep buffers for upload/download and convert to streams and back for processing by the v3 of the custom runner

Reply:
This seems to work:
```typescript
import { Upload } from '@aws-sdk/lib-storage';

storeFile: (filename: string, stream) => {
  if (readOnly) {
    throw new Error('ReadOnly');
  }

  const upload = new Upload({
    client: s3Storage,
    // eslint-disable-next-line @typescript-eslint/naming-convention
    params: { Bucket: bucket, Key: `${prefix}${filename}`, Body: stream },
  });

  return upload.done();
},
```

Reply:
Hmm, is there any distinct speed difference between the two?

Reply:
I didn't really notice any difference, but I only have small projects to test on..